tests/models/test_models.py

```diff
@@ -455,13 +456,23 @@ def test_multihead_head_selection(self):
         for x, y, t in DataLoader(benchmark.train_stream[0].dataset):
             y_mh = model(x, t)
             y_t = model_t0(x)
+
+            # We need to pad y_t to dim with zeros
+            # because y_mh will have max dim of all heads
+            y_t = F.pad(y_t, (0, y_mh.size(1) - y_t.size(1)))
+
```

Comment:
Previous test was assuming that y_t and y_mh are always of the same size. This is not true. The reason why is that when passing a vector as a target to MultiHeadClassifier, it will always return an output that is padded to the maximum size of one of it's head, this is because it assumes that more than one unique target can be inside the vector (unfortunately, most of the time we indeed only pass one task inside the vector, maybe this is a behaviour we want to change). However, because of the bug the output y_t was also of the size of the biggest task because all IncrementalClassifiers were updated on all experiments. Now it's not the case anymore so before checking equality I have to pad y_t with zeros.

Reply:
Correct.